src/styles/settings.css

```diff
@@ -18,267 +33,396 @@ body {
+  flex: 1 1 auto;
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+  overflow: auto;
 }
 
 header {
-  width: 100%;
+  height: 66px;
   display: flex;
-  justify-content: center;
-  box-shadow: inset 0px -1px 1px -1px #bbb;
-  position: relative;
+  align-items: center;
+  flex: 0 0 auto;
+  background: white;
+  box-shadow: rgba(0, 0, 0, 0.13) 0px 2px 6px 0px;
   z-index: 1;
 }
 
-header ul {
-  display: flex;
-  flex-direction: row;
-  list-style: none;
-  height: 48px;
-  padding: 0;
+header h2 {
+  font-size: 16px;
+  font-weight: 400;
   margin: 0;
+  padding: 0 20px;
 }
 
-header .tab-link {
+.tabs {
+  display: flex;
+  overflow: hidden;
+  transition: order 300ms ease-in-out;
+  align-items: stretch;
+}
+
+.tab {
+  display: none;
+  position: relative;
+  flex: 1 0 100%;
+  padding: 20px;
   box-sizing: border-box;
-  border-bottom: 1px solid #eee;
-  padding: 10px 20px;
-  cursor: pointer;
+}
+
+.tab.active {
   display: flex;
-  align-items: center;
-  font-size: 14px;
+  flex-direction: column;
 }
 
-header .active,
-header li:hover {
-  border-bottom: 1px solid red;
+section {
+  margin: 10px;
+  padding: 0 20px;
+  border-radius: var(--m-radius);
+  font-size: var(--font-size);
+  box-shadow: rgba(0, 0, 0, 0.1) 0px 1px 3px 0px;
 }
 
-/* Permissions tab */
+section h3, section .setting {
+  font-size: var(--font-size);
+  font-weight: 500;
+  margin: 0;
+}
 
-.tab-3 .description {
-  width: 40%;
+section label {  
+  display: flex;
+  flex-direction: column !important;
+  align-items: stretch !important;
 }
 
-.container .tab-3 h2 {
-  margin-bottom: 15px;
-  clear: both;
+section button, section .button {
+  margin-top: 5px;
 }
 
-#origins-container {
-  display: inline;
+section .headline {
+  padding: 18px 0;
+  border-bottom: 1px solid var(--border-color);
+  margin-bottom: 18px;
+}
+
+section .field {
+  margin-bottom: 18px;
+}
+
+section .field .option {
+  margin-bottom: 5px;
+}
+
+section .field .description {
+  color: rgb(119, 119, 119);
+}
+
+section .field .subsettings-details {
+  overflow: hidden;
+  height: 0px;
+  margin-left: 30px;
+  transition: height 300ms ease-in-out;
 }
 
-.permissions-toolbar {
-  width: 375px;
+section .field .subsettings-details span {
+  display: block;
 }
 
-#permission-filter {
-  width: 210px;
-  padding-right: 20px;
+section .field .subsettings-details button.test {
+  height: 22px;
+  font-size: 0.8em;
+  margin-left: 15px;
+  padding: 0 15px;
 }
 
-.button-container {
-    display: block;
-    margin-top: 10px;
+section .field .subsettings-details input {
+  margin: 0 4px;
 }
 
-.section:hover .description {
-  visibility: visible;
+section .field--showDetails .subsettings-details {
+  height: auto;
+  padding-top: var(--m-radius);
 }
 
+/* Permissions tab */
+
 /*
  * For styling permissions and custom permissions lists
 */
+
+#origins-container {
+  display: inline;
+}
+
 .origin-list {
-  border: 1px solid #cccaca;
-  list-style: none;
   max-height: 290px;
```

Comment:
Can we change this to `height: 290px;` so the layout doesn't jump on filtering list